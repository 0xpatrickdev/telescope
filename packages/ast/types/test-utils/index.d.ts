import { ProtoType } from '@osmonauts/types';
import { ProtoStore } from '@osmonauts/proto-parser';
import { AminoParseContext } from '../src/encoding/context';
export declare const expectCode: (ast: any) => void;
export declare const printCode: (ast: any) => void;
export declare const prepareContext: (store: ProtoStore, protoFile: string) => {
    context: AminoParseContext;
    root: import("@osmonauts/types").ProtoRoot;
    protos: ProtoType[];
};
