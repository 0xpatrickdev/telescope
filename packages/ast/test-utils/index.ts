import generate from '@babel/generator';
import { ProtoStore, getNestedProto, parseProto } from '@osmonauts/proto-parser'
import { ProtoType } from '../src/encoding/proto/types';
import { ParseContext } from '../src/encoding/context';

export const expectCode = (ast) => {
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}
export const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}

export const prepareContext = (store: ProtoStore, protoFile: string) => {
    const ref = store.findProto(protoFile);
    const traversed = ref.traversed
    if (!traversed) {
        throw new Error('not traversed!');
    }
    const proto: Record<string, ProtoType> = getNestedProto(traversed);
    const protos: ProtoType[] = Object.values(proto).filter(
        proto => proto.name.startsWith('Msg')
            &&
            !proto.name.endsWith('Response')
            &&
            proto.name !== 'Msg'
    );

    const context = new ParseContext(ref, store);

    return {
        context,
        root: traversed,
        protos
    };
};
