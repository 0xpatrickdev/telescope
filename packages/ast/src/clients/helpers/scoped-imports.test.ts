import { expectCode, printCode } from '../../../test-utils'

import {
    recursiveObjectProps,
    lcdArguments,
    createScopedImportObject
} from './scoped-imports';

it('recursiveObjectProps', async () => {
    expectCode(recursiveObjectProps(['cosmos', 'bank', 'v1beta1']))
});

it('createScopedImportObject', async () => {
    expectCode(createScopedImportObject({
        cosmos: {
            bank: {
                v1beta1: "./proto/cosmos/bank/v1beta1/query.lcd"
            },
            gov: {
                v1beta1: "./proto/cosmos/bank/v1beta1/query.lcd"
            },
        },
        osmosis: {
            gamm: {
                v1beta1: "./proto/cosmos/bank/v1beta1/query.lcd"
            }
        }
    },
        'lcd',
        'LCDQueryClient',
        lcdArguments()
    ))
});
