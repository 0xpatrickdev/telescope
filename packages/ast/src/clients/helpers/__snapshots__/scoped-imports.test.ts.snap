// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createScopedImportObject 1`] = `
"export const lcd = async ({
  restEndpoint
}) => ({
  cosmos: {
    bank: {
      v1beta1: new (await import(\\"./proto/cosmos/bank/v1beta1/query.lcd\\")).LCDQueryClient({
        restEndpoint
      })
    },
    gov: {
      v1beta1: new (await import(\\"./proto/cosmos/bank/v1beta1/query.lcd\\")).LCDQueryClient({
        restEndpoint
      })
    }
  },
  osmosis: {
    gamm: {
      v1beta1: new (await import(\\"./proto/cosmos/bank/v1beta1/query.lcd\\")).LCDQueryClient({
        restEndpoint
      })
    }
  }
});"
`;

exports[`recursiveObjectProps 1`] = `
"{
  cosmos: {
    bank: {
      v1beta1: v1beta1
    }
  }
}"
`;
